http/oidc/src/main/java/org/wildfly/security/http/oidc/OidcRequestAuthenticator.java

```diff
@@ -222,7 +229,8 @@ protected String getRedirectUri(String state) {
             String redirectUri = rewrittenRedirectUri(url);
             URIBuilder redirectUriBuilder = new URIBuilder(deployment.getAuthUrl());
             redirectUriBuilder.addParameter(RESPONSE_TYPE, CODE)
-                    .addParameter(CLIENT_ID, deployment.getResourceName());
+                    .addParameter(CLIENT_ID, deployment.getResourceName())
+                    .addParameter(NONCE, String.valueOf(sessionId.hashCode()));
```

Comment:
Here, we need to use a cryptographic hash of the random value. We can use `MessageDigest` for this.

Reply:
done